runtime/src/main/java/org/corfudb/runtime/collections/ProtobufIndexer.java

```diff
@@ -0,0 +1,65 @@
+package org.corfudb.runtime.collections;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import com.google.protobuf.Message;
+
+import org.corfudb.runtime.CorfuOptions;
+import org.corfudb.runtime.collections.CorfuTable.Index;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Optional;
+
+/**
+ * Created by hisundar on 2019-08-12.
+ */
+public class ProtobufIndexer implements CorfuTable.IndexRegistry<Message, CorfuRecord<Message, Message>> {
+
+    private HashMap<String,
+            CorfuTable.Index<Message, CorfuRecord<Message, Message>, ? extends Comparable<?>>>
+            indices = new HashMap<>();
+
+    ProtobufIndexer(Message payloadSchema) {
+        payloadSchema.getDescriptorForType().getFields().forEach(this::registerIndices);
+    }
+
+    private <T extends Comparable<T>> CorfuTable.Index<Message, CorfuRecord<Message, Message>, ? extends Comparable<?>>
+    getIndex(String indexName, FieldDescriptor fieldDescriptor) {
+
+        return new Index<>(
+                () -> indexName,
+                (CorfuTable.IndexFunction<Message, CorfuRecord<Message, Message>, T>)
+                        (key, val) -> (T) val.getPayload().getField(fieldDescriptor));
+    }
+
+    void registerIndices(final Descriptors.FieldDescriptor fieldDescriptor) {
+
+        if (fieldDescriptor.getOptions().getExtension(CorfuOptions.schema).getSecondaryKey()) {
+            final String indexName = fieldDescriptor.getName();
+            if (fieldDescriptor.getType() == FieldDescriptor.Type.GROUP) {
+                throw new IllegalArgumentException("group is a deprecated, unsupported type");
```

Comment:
why not return `Result<Exception, List<Index>>` instead of throwing runtime exception?